pkg/destroy/gcp/disk.go

```diff
@@ -27,10 +28,10 @@ func (o *ClusterUninstaller) listDisksWithFilter(fields string, filter string, f
 		req = req.Filter(filter)
 	}
 	err := req.Pages(ctx, func(list *compute.DiskAggregatedList) error {
-		for _, scopedList := range list.Items {
+		for key, scopedList := range list.Items {
 			for _, item := range scopedList.Disks {
 				if filterFunc == nil || filterFunc != nil && filterFunc(item) {
-					zone := o.getZoneName(item.Zone)
+					zone := strings.Split(key, "/")[1]
```

Comment:
Instead of this, let's update getZoneName to split based on /projects/ similar to how you did in instance.go. This allows us to update it in a single place if it changes again in the future.

`Split(instanceURL, "/projects/")[1]`

https://github.com/openshift/installer/blob/a7f8c8da45993927639ca16c725e04d9000bdb0f/pkg/destroy/gcp/gcp.go#L168-L175

Reply:
Would split based on /projects/ not cause a problem? I was worried that there might be something else named projects in the URL. That might cause some problems.

Reply:
There might be a risk of that... probably less of a risk with the surrounding slashes. I also notice that the zone name is the last entity in strings.Split(path, "/").... so that might be better here.
```
parts:=strings.Split(path, "/")
if len(parts) {
  return parts[len(parts)-1]
}
```